[2024-10-10T16:30:25.117+0900] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-10T16:30:25.133+0900] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: handwriting_dag.preprocessing_new_data manual__2024-10-10T07:24:54.332251+00:00 [queued]>
[2024-10-10T16:30:25.140+0900] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: handwriting_dag.preprocessing_new_data manual__2024-10-10T07:24:54.332251+00:00 [queued]>
[2024-10-10T16:30:25.141+0900] {taskinstance.py:2865} INFO - Starting attempt 2 of 2
[2024-10-10T16:30:25.153+0900] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): preprocessing_new_data> on 2024-10-10 07:24:54.332251+00:00
[2024-10-10T16:30:25.170+0900] {standard_task_runner.py:72} INFO - Started process 57839 to run task
[2024-10-10T16:30:25.182+0900] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'handwriting_dag', 'preprocessing_new_data', 'manual__2024-10-10T07:24:54.332251+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/model_dag.py', '--cfg-path', '/var/folders/tx/bcbxk5bj33x4r2rt041m7bmr0000gn/T/tmpgpks_sw6']
[2024-10-10T16:30:25.184+0900] {standard_task_runner.py:105} INFO - Job 15: Subtask preprocessing_new_data
[2024-10-10T16:30:25.238+0900] {task_command.py:467} INFO - Running <TaskInstance: handwriting_dag.preprocessing_new_data manual__2024-10-10T07:24:54.332251+00:00 [running]> on host hyowon-kim-macbook-pro.local
[2024-10-10T16:30:25.312+0900] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='handwriting_dag' AIRFLOW_CTX_TASK_ID='preprocessing_new_data' AIRFLOW_CTX_EXECUTION_DATE='2024-10-10T07:24:54.332251+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-10T07:24:54.332251+00:00'
[2024-10-10T16:30:25.314+0900] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-10-10T16:30:25.315+0900] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-10-10T16:30:25.315+0900] {logging_mixin.py:190} INFO - Current task name:preprocessing_new_data state:running start_date:2024-10-10 07:30:25.133908+00:00
[2024-10-10T16:30:25.316+0900] {logging_mixin.py:190} INFO - Dag name:handwriting_dag and current dag run status:running
[2024-10-10T16:30:25.316+0900] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-10T16:30:25.337+0900] {model_dag.py:60} INFO - Processed image: 5.png
[2024-10-10T16:30:25.338+0900] {model_dag.py:60} INFO - Processed image: 2.png
[2024-10-10T16:30:25.339+0900] {model_dag.py:60} INFO - Processed image: 3.png
[2024-10-10T16:30:25.340+0900] {model_dag.py:60} INFO - Processed image: 1.png
[2024-10-10T16:30:25.341+0900] {model_dag.py:67} INFO - Processed 4 new images
[2024-10-10T16:30:25.341+0900] {python.py:240} INFO - Done. Returned value was: (tensor([[[[-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          ...,
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.]]],


        [[[-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          ...,
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.]]],


        [[[-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          ...,
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.]]],


        [[[-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          ...,
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.],
          [-1., -1., -1.,  ..., -1., -1., -1.]]]]), tensor([5, 2, 3, 1]))
[2024-10-10T16:30:25.353+0900] {xcom.py:690} ERROR - Object of type tuple is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your airflow config or make sure to decorate your object with attr.
[2024-10-10T16:30:25.355+0900] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/json.py", line 92, in default
    return serialize(o)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 151, in serialize
    return encode(classname or serialized_classname, version, serialize(data, depth + 1))
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 128, in serialize
    return [serialize(d, depth + 1) for d in o]
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 128, in <listcomp>
    return [serialize(d, depth + 1) for d in o]
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 190, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'torch.Tensor'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 789, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session_or_null)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3638, in xcom_push
    XCom.set(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/api_internal/internal_api_call.py", line 139, in wrapper
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/xcom.py", line 249, in set
    value = cls.serialize_value(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/xcom.py", line 688, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/Cellar/python@3.10/3.10.15/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/json.py", line 103, in encode
    o = self.default(o)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/json.py", line 94, in default
    return super().default(o)
  File "/usr/local/Cellar/python@3.10/3.10.15/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type tuple is not JSON serializable
[2024-10-10T16:30:25.362+0900] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-10-10T16:30:25.363+0900] {logging_mixin.py:190} INFO - Task start:2024-10-10 07:30:25.133908+00:00 end:2024-10-10 07:30:25.362065+00:00 duration:0.228157
[2024-10-10T16:30:25.364+0900] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): preprocessing_new_data> dag:<DAG: handwriting_dag> dagrun:<DagRun handwriting_dag @ 2024-10-10 07:24:54.332251+00:00: manual__2024-10-10T07:24:54.332251+00:00, state:running, queued_at: 2024-10-10 07:24:54.348490+00:00. externally triggered: True>
[2024-10-10T16:30:25.364+0900] {logging_mixin.py:190} INFO - Failure caused by Object of type tuple is not JSON serializable
[2024-10-10T16:30:25.365+0900] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=handwriting_dag, task_id=preprocessing_new_data, run_id=manual__2024-10-10T07:24:54.332251+00:00, execution_date=20241010T072454, start_date=20241010T073025, end_date=20241010T073025
[2024-10-10T16:30:25.375+0900] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-10T16:30:25.376+0900] {standard_task_runner.py:124} ERROR - Failed to execute job 15 for task preprocessing_new_data (Object of type tuple is not JSON serializable; 57839)
Traceback (most recent call last):
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/json.py", line 92, in default
    return serialize(o)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 151, in serialize
    return encode(classname or serialized_classname, version, serialize(data, depth + 1))
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 128, in serialize
    return [serialize(d, depth + 1) for d in o]
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 128, in <listcomp>
    return [serialize(d, depth + 1) for d in o]
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 190, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'torch.Tensor'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 789, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session_or_null)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3638, in xcom_push
    XCom.set(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/api_internal/internal_api_call.py", line 139, in wrapper
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/xcom.py", line 249, in set
    value = cls.serialize_value(
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/models/xcom.py", line 688, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/Cellar/python@3.10/3.10.15/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/json.py", line 103, in encode
    o = self.default(o)
  File "/Users/alookso/1007MLpj/myenv/lib/python3.10/site-packages/airflow/utils/json.py", line 94, in default
    return super().default(o)
  File "/usr/local/Cellar/python@3.10/3.10.15/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type tuple is not JSON serializable
[2024-10-10T16:30:25.413+0900] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-10T16:30:25.432+0900] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-10T16:30:25.432+0900] {local_task_job_runner.py:245} INFO - ::endgroup::
